Vagrantfile

```diff
@@ -52,13 +53,21 @@ Vagrant.configure("2") do |config|
 # Fix gnome-terminal
     sudo localectl set-locale LANG="en_GB.utf8"
 
+# Docker
+    sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
+    sudo apt-add-repository 'deb https://apt.dockerproject.org/repo ubuntu-xenial main'
+    sudo apt-get update
+    sudo apt-get install -y docker-engine
+    sudo usermod -aG docker vagrant
+
 # Verify xapi can be built
     git clone git://github.com/xapi-project/xen-api
     eval `opam config env`
-    cd xen-api; ./configure; make; make test; cd ..
+    cd xen-api; ./configure; make; make test > test.log; cd ..
```

Comment:
It's no reason to stop the merge for this, so am going to merge now, but it's worth thinking about later.